src/psc_datalogger/gui.py

```diff
@@ -129,12 +129,19 @@ def handle_start_stop(self, checked):
     def handle_instrument_changed(self):
         """Handle when any of the instrument configurations change"""
         for i in self.instruments:
-            self.connection_manager.set_instrument(
-                i.instrument_number,
-                i.isChecked(),
-                i.get_address(),
-                i.get_temperature_checked(),
-            )
+            try:
+                self.connection_manager.set_instrument(
+                    i.instrument_number,
+                    i.isChecked(),
+                    i.get_address(),
+                    i.get_temperature_checked(),
+                )
+            except ValueError:
+                # Expected when first activating an instrument; the Address field
+                # will be empty
+                logging.exception(
+                    f"ValueError when initializing instrument {i.instrument_number}"
```

Comment:
This is an expected error, which is always captured and handled appropriately, but the Exception message here is not very useful.

Also, logging.exception will flag up this message as an error, but above you state that it is just the normal result when first activating an instrument? This should be INFO level or below.

Reply:
I've ended up downgrading this to a `warning`, but moving it down to `info` would mean it wouldn't appear on the default logging level configuration. It's also possible that this error could be coming from something else and not just that expected initialization error; hence why I do want it to still be visible.

Changed in 12288e0aab432a492f497cddbf445f9c2dd1fce5